### ISSUE TITLE
CIAB: edge server returns "Cycle Prohibited" after add a new Delivery Service

### ISSUE THREAD

**Issue by Shihta** [apache/trafficcontrol#3215]
I built a CIAB environment by using the latest commit (5ca7cef3ec6b6625e1bdfda962edef40c69c40fd)

My steps are shown as follows and most of them are operated via portal:

1. Create a Delivery Service
![2019-01-15 2 57 54](https://user-images.githubusercontent.com/5186176/51165081-09ed6200-18da-11e9-871a-069bb3241343.png)
![2019-01-15 3 00 16](https://user-images.githubusercontent.com/5186176/51165195-63ee2780-18da-11e9-9dc9-68e7576748b2.png)

2. Assign Delivery Services to edge
![2019-01-15 3 01 59](https://user-images.githubusercontent.com/5186176/51165247-8d0eb800-18da-11e9-841d-122596ac9910.png)

3. take snapshot
![2019-01-15 3 02 33](https://user-images.githubusercontent.com/5186176/51165347-df4fd900-18da-11e9-9a3f-168a755d3930.png)

4. update config of edge server by execute: `traffic_ops_ort -k BADASS INFO https://trafficops.infra.ciab.test:443 admin:twelve`

After the environment are all set, I test it by **curl**. The result is shown as follows:
```
$ curl -I -L http://cdn.key.mycdn.ciab.test
HTTP/1.1 302 Found
Location: http://edge.key.mycdn.ciab.test/
Transfer-Encoding: chunked
Date: Tue, 15 Jan 2019 07:07:11 GMT

HTTP/1.1 400 Multi-Hop Cycle Detected
Date: Tue, 15 Jan 2019 07:07:11 GMT
Via: http/1.1 mid.infra.ciab.test (ApacheTrafficServer/7.1.4 [uEcMs f p eH:t cCMi p s ]), http/1.1 edge.infra.ciab.test (ApacheTrafficServer/7.1.4 [uScMs f p eN:t cCMi pSs ]), http/1.1 mid.infra.ciab.test (ApacheTrafficServer/7.1.4 [uScMs f p eN:t cCMi p sS]), http/1.1 edge.infra.ciab.test (ApacheTrafficServer/7.1.4 [uScMs f p eN:t cCMi pSs ])
Server: ATS/7.1.4
Cache-Control: no-store
Content-Type: text/html
Content-Language: en
Content-Length: 347
Age: 0
Connection: keep-alive
```

Any idea?
Thanks!

**Comment by ocket8888** [comment]
It is a known issue that having multiple Delivery Services with the same origin FQDN causes undefined behavior. A database having two Delivery Services with the same origin FQDN is a database that has bad data, so I wouldn't expect it to work properly. There's an effort ongoing to explicitly prohibit putting the system into this known, bad state, but we're still debating specifics and timing.

**Comment by Shihta** [comment]
It has been fixed by #3252
